Modernize phylogenetic build [#2]

genehack opened the pull request:
## Description of proposed changes

Updates yellow-fever phylo build to match pathogen-repo-guide standard.

Also updates various ingest curation steps to use `augur curate` instead of vendored ingest scripts.

## Related issue(s)

#2 #8

<!--
Link any related issues here. Use GitHub's special keywords if appropriate¹.
Type `#` followed the name of an issue and GitHub will auto-suggest the issue number for you.

¹ https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/using-keywords-in-issues-and-pull-requests
-->

## Checklist

<!--
Make sure checks are successful at the bottom of the PR.

If applicable, add:
- any changes to existing tests
- any additional manual testing to confirm changes

Please add a note if you need help with adding tests.
-->

- [ ] Checks pass

<!-- 🙌 Thank you for contributing to Nextstrain! ✨ -->

genehack committed:
Remove `_LEGACY` [#2]

Also remove exclusion from .pre-commit-config.yaml

genehack committed:
Misc cleanups and formatting [#2]

* Drop to Python 3.11 in GitHub pre-commit CI, to match actual
  development practice
* Clean up formatting of several files
* Convert all Markdown links to use reference dictionary at end of
  file
* Add `log` and `benchmark` to all Snakemake rules that didn't have
  them
* Remove `dump_ncbi_dataset_report` rule

genehack committed:
Add `.snakemake` to `clean` rules [#2]

genehack committed:
Note why we're wildcarding a single thing [#2]

joverlee521 commented on ingest/rules/curate.smk:
Did ingest changes sneak in here on accident or were they intended? 
If intended, might as well tack on all changes for https://github.com/nextstrain/yellow-fever/issues/8.

genehack replied:
They were intended but are limited to whitespace and formatting changes. I was going to do #8 on top of these.

joverlee521 replied:
Ah, it currently includes migrating from `./vendored/transform-genbank-location` to `augur curate parse-genbank-location`.

genehack replied:
ugh, okay

genehack replied:
done!

genehack committed:
Phylogenetic workflow modernization [#2]

* Update template files to remove instructions and placeholders
* Add Genbank and GFF references
* Bring in phylo workflow from pathogen-repo-guide and modify to work

genehack committed:
Update ingest to new `augur curate` commands [#8]

Also adds logging redirects everywhere

genehack committed:
git subrepo pull (merge) ingest/vendored

subrepo:
  subdir:   "ingest/vendored"
  merged:   "980c010"
upstream:
  origin:   "https://github.com/nextstrain/ingest"
  branch:   "main"
  commit:   "258ab8c"
git-subrepo:
  version:  "0.4.7"
  origin:   "https://github.com/ingydotnet/git-subrepo/"
  commit:   "c9ec2a6"

joverlee521 commented on phylogenetic/rules/prepare_sequences.smk:
```diff
@@ -1,22 +1,68 @@
+    input:
+        exclude = config["files"]["exclude"],
+        include = config["files"]["include_genome"],
+        metadata = "../ingest/results/metadata.tsv",
+        sequences = "../ingest/results/sequences.fasta"
```
_non-blocking_

Once the ingest/phylogenetic workflows are fully automated, I would avoid accessing `../ingest` since the usual pattern is for people to start from public hosted files on S3. 

(I totally get that this is nice for testing while the data is not up on S3 yet!)

genehack replied:
Added a TODO comment to this effect.

joverlee521 commented on ingest/rules/curate.smk:
```diff
@@ -18,6 +18,8 @@ rule fetch_general_geolocation_rules:
         general_geolocation_rules="data/general-geolocation-rules.tsv",
     params:
         geolocation_rules_url=config["curate"]["geolocation_rules_url"],
+    log:
```
+1 for more use of `log` in Snakemake workflows, but the log file is empty unless you redirect stdout/stderr to the file in the shell command. 

See example in [Nextstrain Snakemake guide](https://docs.nextstrain.org/en/latest/reference/snakemake-style-guide.html#log-standard-out-and-error-output-to-log-files-and-the-terminal)

genehack replied:
:facepalm:

fixed throughout.

joverlee521 commented on ingest/Snakefile:
```diff
@@ -36,6 +36,7 @@ else:
 rule clean:
     params:
         targets = [
+            ".snakemake",
```
Nice, makes sense if you want to nuke everything and start the workflow from scratch!

joverlee521 reviewed (approved):
Reviewed workflow bits only, will leave scientific review to others.

genehack committed:
add a TODO based on PR feedback [#2]

genehack committed:
Refinements to phylogenetic build [#2]

* Switch Auspice to default color-by=region
* Remove extra `metadata_columns` (and associated config) when calling
  `augur export v2`
* Move `--columns` arg of `augur traits` into config
* Add `--metadata-id-columns` arg to `augur traits` invocation --
  fixes issue where traits were being improperly inferred
* Use a mid-point tree rooting
